mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java

```diff
@@ -69,22 +82,221 @@
+        mailboxPathV2DAO = spy(new CassandraMailboxPathV2DAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION));
         CassandraUserMailboxRightsDAO userMailboxRightsDAO = new CassandraUserMailboxRightsDAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+        aclMapper = spy(new CassandraACLMapper(
+            cassandra.getConf(),
+            new CassandraUserMailboxRightsDAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION),
+            CassandraConfiguration.DEFAULT_CONFIGURATION));
         testee = new CassandraMailboxMapper(
             mailboxDAO,
             mailboxPathDAO,
             mailboxPathV2DAO,
             userMailboxRightsDAO,
-            new CassandraACLMapper(cassandra.getConf(),
-                new CassandraUserMailboxRightsDAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION),
-                CassandraConfiguration.DEFAULT_CONFIGURATION));
+            aclMapper);
+    }
+
+    @Nested
+    class ConsistencyTest {
+
+        private CassandraId inboxId;
+        private MailboxPath inboxPath;
+        private Mailbox inbox;
+        private MailboxPath inboxPathRenamed;
+        private Mailbox inboxRenamed;
+        private MailboxQuery.UserBound inboxSearchQuery;
+        private MailboxQuery.UserBound inboxRenamedSearchQuery;
+
+        @BeforeEach
+        void setUp() {
+            inboxId = CassandraId.timeBased();
+            inboxPath = MailboxPath.forUser(USER, INBOX);
+            inbox = new Mailbox(inboxPath, UID_VALIDITY, inboxId);
+
+            inboxPathRenamed = MailboxPath.forUser(USER, INBOX_RENAMED);
+            inboxRenamed = new Mailbox(inboxPathRenamed, UID_VALIDITY, inboxId);
+            inboxSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPath)
+                .expression(new ExactName(INBOX))
+                .build()
+                .asUserBound();
+            inboxRenamedSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPathRenamed)
+                .expression(new ExactName(INBOX_RENAMED))
+                .build()
+                .asUserBound();
+        }
+
+        @Test
+        void saveOnCreateShouldBeConsistentWhenFailToPersistsMailbox() {
+            doReturn(Mono.error(new RuntimeException("mock exception")))
+                .when(mailboxDAO)
+                .save(inbox);
+
+            doQuietly(() -> testee.save(inbox));
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThatThrownBy(() -> testee.findMailboxById(inboxId))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThatThrownBy(() -> testee.findMailboxByPath(inboxPath))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .isEmpty();
```

Comment:
Just a question, I'm not sure of the answer myself: how can we know that the mailboxPath exists in this case but not the mailbox? If the path doesn't exist I guess you get MailboxNotFoundException thrown too?

Reply:
the `findMailboxWithPathLike()` returns a List. So, in case of notfound, it actually return an empty list :)

Reply:
But doesn't it return an empty list in both cases as well if I understand? 
* I have a mailboxPath without its mailbox -> empty list
* I don't have mailboxPath and mailbox -> empty list too?

Reply:
We had a discussion, I focused on the assertions at mapper level and ignored the inconsistencies at the lower level (DAO). With the statement of me, as long as the mapper working fine, we should be fine. 

At the DAO level, some inconsistencies are non-critical to the mapper - mapper tests still pass, and some are critical to the mapper - mapper tests fail.